clang/test/SemaSYCL/attr-add-ir-attributes-merge.cpp

```diff
@@ -0,0 +1,53 @@
+// RUN: %clang_cc1 -fsycl-is-device -std=gnu++11 -fsyntax-only -verify %s
+
+// Tests that add_ir_attributes_* allows merging when the filter and IR
+// attributes to generate are the same.
+
+void FunctionRedecl1();
+[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl1();
+[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl1();
```

Comment:
Can you add a test where names are the same but values are different  

E.g. 

[`[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, false)]] void FunctionRedecl1();
`